demos/security_barrier_camera_demo/main.cpp

```diff
@@ -65,22 +65,19 @@ struct BboxAndDescr {
 };
 
 struct InferRequestsContainer {
-    explicit InferRequestsContainer(std::vector<InferRequest> inferRequests):
-        actualInferRequests{inferRequests} {
-        for (auto& ir : actualInferRequests) {
-            inferRequests.push_back(ir);
-        }
-    }
     InferRequestsContainer() = default;
-    InferRequestsContainer& operator=(const InferRequestsContainer& other) {  // copy assignment
-        if (this != &other) {  // self-assignment check expected
-            this->actualInferRequests = other.actualInferRequests;
-            for (auto& ir : this->actualInferRequests) {
-                this->inferRequests.container.push_back(ir);
-            }
+    InferRequestsContainer(const InferRequestsContainer&) = delete;
+    InferRequestsContainer& operator=(const InferRequestsContainer&) = delete;
+
+    void assign(const std::vector<InferRequest>& inferRequests) {
+        actualInferRequests = std::move(inferRequests);
```

Comment:
```suggestion
        actualInferRequests = inferRequests;
```
There is no `operator=( const vector&& other )` for `vector`. And `operator=( const vector& other );` will be used any way.

Reply:
Good catch; fixed. It must've been a leftover from an earlier approach I tried.